src/chainstate/stacks/db/blocks.rs

```diff
@@ -2702,6 +2747,103 @@ impl StacksChainState {
 
         Ok(ret)
     }
+
+    pub fn will_admit_mempool_tx<R: Read>(&mut self, current_burn: &BurnchainHeaderHash, current_block: &BlockHeaderHash, fd: &mut R) -> Result<StacksTransaction, MemPoolRejection> {
+        // 1: it should parse as a tx.
+        let mut fd_counter = ReadCounter::from_reader(fd);
```

Comment:
This check should also ensure that the transaction isn't too big (i.e. no more than `MAX_MESSAGE_LEN`, or a similarly-defined constant for transactions).

Reply:
Addressed in ca8eacd1b -- I added a bound reader to the deserialization method of `StacksTransaction`

Reply:
Thanks!